TST: test_bounds_gauss2d_lsq[TRFLSQFitter] fails with scipy 1.11.2

Looks like a new failure from scipy 1.11.2 released today.

```
Scipy: 1.11.2
...
_______________ TestBounds.test_bounds_gauss2d_lsq[TRFLSQFitter] _______________

self = <astropy.modeling.tests.test_constraints.TestBounds object at 0x7f28ad9105d0>
fitter = <astropy.modeling.fitting.TRFLSQFitter object at 0x7f2877fb8590>

    @pytest.mark.parametrize("fitter", fitters)
    def test_bounds_gauss2d_lsq(self, fitter):
        fitter = fitter()
...
            with ctx, ctx2:
>               model = fitter(gauss, X, Y, self.data)

astropy/modeling/tests/test_constraints.py:218: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/modeling/fitting.py:301: in wrapper
    return func(self, model, x, y, z=z, **kwargs)
astropy/modeling/fitting.py:1366: in __call__
    init_values, fitparams, cov_x = self._run_fitter(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <astropy.modeling.fitting.TRFLSQFitter object at 0x7f2877fb8590>
model = <Gaussian2D(amplitude=41.49392764, x_mean=4.89282467, y_mean=5.07465274, x_stddev=4., y_stddev=4., theta=0.5)>
farg = (<Gaussian2D(amplitude=41.49392764, x_mean=4.89282467, y_mean=5.07465274, x_stddev=4., y_stddev=4., theta=0.5)>, None,...
         478.,  535.],
       [ 443.,  416.,  439.,  472.,  472.,  492.,  523.,  569.,  487.,
         441.,  428.]]))
maxiter = 100, acc = 1e-07, epsilon = 1.4901161193847656e-08
estimate_jacobian = False

    def _run_fitter(self, model, farg, maxiter, acc, epsilon, estimate_jacobian):
...
        fitter_to_model_params(model, self.fit_info.x, False)
        if not self.fit_info.success:
>           warnings.warn(
                f"The fit may be unsuccessful; check: \n    {self.fit_info.message}",
                AstropyUserWarning,
            )
E           astropy.utils.exceptions.AstropyUserWarning: The fit may be unsuccessful; check: 
E               The maximum number of function evaluations is exceeded.

astropy/modeling/fitting.py:1548: AstropyUserWarning
```

Example log: https://github.com/astropy/astropy/actions/runs/5898301068/job/15999160283

--- a/astropy/modeling/tests/test_constraints.py
+++ b/astropy/modeling/tests/test_constraints.py
@@ -14,9 +14,12 @@ from numpy.testing import assert_allclose
 from astropy.modeling import fitting, models
 from astropy.modeling.core import Fittable1DModel
 from astropy.modeling.parameters import Parameter
+from astropy.utils import minversion
 from astropy.utils.compat.numpycompat import NUMPY_LT_2_0
 from astropy.utils.compat.optional_deps import HAS_SCIPY
 from astropy.utils.exceptions import AstropyUserWarning
+
+SCIPY_LT_1_11_2 = not minversion("scipy", "1.11.2") if HAS_SCIPY else True
 
 fitters = [
     fitting.LevMarLSQFitter,
@@ -208,7 +211,7 @@ class TestBounds:
             ctx2 = nullcontext()
             if isinstance(fitter, fitting.TRFLSQFitter):
                 ctx = np.errstate(invalid="ignore", divide="ignore")
-                if not NUMPY_LT_2_0:
+                if not NUMPY_LT_2_0 or not SCIPY_LT_1_11_2:
                     ctx2 = pytest.warns(
                         AstropyUserWarning, match="The fit may be unsuccessful"
                     )

